File: src/main/scala/pl/edu/pw/compression/inter/LZW.scala
package pl.edu.pw.compression.inter

import java.io.{InputStream, OutputStream}

import pl.edu.pw.compression.LZWCompression


/**
  * Created by clutroth on 01.12.16.
  */
object LZW {
  val BUF_SIZE = 1 << 10

  def decompress(from: InputStream, to: OutputStream, dictionary: Array[Byte]) =
    ???

  def compress(from: InputStream, to: OutputStream, dictionary: Array[Byte]) = {
    def firstCharacter = {
      val b = Array[Byte](1)
      from.read(b)
      b.head
    }
    val comp = LZWCompressionImpl.begin(dictionary, firstCharacter)
    val buf = new Array[Byte](BUF_SIZE)
    def readBuffer(buf:Array[Byte]) ={
if(from.)
    }
  }

}

File: src/main/java/pl/edu/pw/compression/LZW.java
package pl.edu.pw.compression;


import pl.edu.pw.compression.inter.LZWDecompressionImpl;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.Arrays;
import java.util.List;

/**
 * Serdecznie polecam https://pl.wikipedia.org/wiki/LZW
 * Created by clutroth on 01.12.16.
 */
public class LZW {

    public OutputStream decompress(InputStream is, OutputStream os, byte[] dictionary) {
        LZWDecompression dec = LZWDecompressionImpl.begin("abcd".getBytes(), 4);
        LZWDecompression nowszy = dec.put(2);

        List<Byte> zakodowanaWartośćDoZapisu = (List<Byte>) nowszy.value(); //To jest optional, tłumaczony na listę
        LZWDecompression jeszczenowszy = dec.put(33);
        LZWDecompression jeszczejeszczenowszy = dec.put(62);
        LZWDecompression jeszczejeszczejeszczenowszy = dec.put(82);


        return null;
    }

    public OutputStream compress(InputStream is, OutputStream os, byte[] dictionary) {
        return null;
    }

}
